src/sparse/node.rs

```diff
@@ -30,21 +30,22 @@ pub(crate) struct Node {
 }
 
 impl Node {
-    pub fn create_leaf(key: &[u8], data: &[u8]) -> Self {
+    pub fn create_leaf(key: &Bytes32, data: &[u8]) -> Self {
         let buffer = Self::default_buffer();
         let mut node = Self { buffer };
         node.set_height(0);
-        node.set_bytes_prefix(&[LEAF]);
-        node.set_bytes_lo(&sum(key));
+        node.set_prefix(LEAF);
+        node.set_bytes_lo(key);
         node.set_bytes_hi(&sum(data));
         node
     }
 
     pub fn create_node(left_child: &Node, right_child: &Node) -> Self {
         let buffer = Self::default_buffer();
         let mut node = Self { buffer };
-        node.set_height(left_child.height() + 1);
-        node.set_bytes_prefix(&[NODE]);
+        let height = std::cmp::max(left_child.height(), right_child.height()) + 1;
```

Comment:
This looks like an orthogonal change?

Reply:
Yes I may undo this change (and a couple other changes) here, but it was a necessary change in the SMT PR (I just brought the code over directly).